code/ui/components/src/components/Button/Button.tsx

```diff
@@ -1,29 +1,121 @@
   tertiary?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   gray?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   inForm?: boolean;
-  disabled?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   small?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   outline?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   containsIcon?: boolean;
-  children?: ReactNode;
-  href?: string;
-}>(
-  ({ small, theme }) => ({
+}
+
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ asChild, onClickAnimation = 'none', onClick, ...props }, ref) => {
+    const Comp = asChild ? Slot : 'button';
+    let { variant, size } = props;
+
+    const [isAnimating, setIsAnimating] = useState(false);
+
+    const handleClick = (event: SyntheticEvent) => {
+      if (onClick) onClick(event);
+      if (onClickAnimation === 'none') return;
+      setIsAnimating(true);
+    };
+
+    useEffect(() => {
+      const timer = setTimeout(() => {
+        if (isAnimating) setIsAnimating(false);
+      }, 1000);
+      return () => clearTimeout(timer);
+    }, [isAnimating]);
+
+    // Match the old API with the new API
+    // TODO: Remove this after 9.0
+    if (props.primary) {
+      variant = 'solid';
+      size = 'medium';
+    }
+
+    // Match the old API with the new API
+    // TODO: Remove this after 9.0
+    if (props.secondary || props.tertiary || props.gray || props.outline || props.inForm) {
+      variant = 'outline';
+      size = 'medium';
+    }
+
+    return (
+      <StyledButton
+        as={Comp}
+        ref={ref}
+        variant={variant}
+        size={size}
+        onClick={handleClick}
+        isAnimating={isAnimating}
+        animation={onClickAnimation}
+        {...props}
+      />
+    );
+  }
+);
+
+Button.displayName = 'Button';
+
+const StyledButton = styled.button<
+  ButtonProps & {
+    isAnimating: boolean;
+    animation: ButtonProps['onClickAnimation'];
+  }
+>(
+  ({
+    theme,
+    variant = 'outline',
+    size = 'small',
+    disabled = false,
+    active = false,
+    isAnimating,
+    animation,
+    padding = 'medium',
+  }) => ({
     border: 0,
-    borderRadius: '3em',
-    cursor: 'pointer',
-    display: 'inline-block',
+    cursor: disabled ? 'not-allowed' : 'pointer',
+    display: 'inline-flex',
+    gap: '6px',
+    alignItems: 'center',
+    justifyContent: 'center',
     overflow: 'hidden',
-    padding: small ? '8px 16px' : '13px 20px',
+    padding: `${(() => {
+      if (padding === 'small' && size === 'small') return '0 7px';
+      if (padding === 'small' && size === 'medium') return '0 9px';
+      if (size === 'small') return '0 10px';
+      if (size === 'medium') return '0 12px';
+      return 0;
+    })()}`,
```

Comment:
you don't need to wrap this in a string here.

```suggestion
    padding: (() => {
      if (padding === 'small' && size === 'small') return '0 7px';
      if (padding === 'small' && size === 'medium') return '0 9px';
      if (size === 'small') return '0 10px';
      if (size === 'medium') return '0 12px';
      return 0;
    })(),
```

Reply:
Fixed it